File: java/07 - Demo 3 - Refactoring/src/test/java/movemethod/OverdraftChargeTest.java
package movemethod;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class OverdraftChargeTest {

    @Test
    public void overdraftCharge() {
        Account account = new Account(new AccountType(false), 20);
        double result = account.bankcharge();
        assertEquals(39.5, result, 0.0);
    }

    @Test
    public void overdraftChargeForPremiumAccountIs10() {
        Account account = new Account(new AccountType(true), 5);
        double result = account.bankcharge();
        assertEquals(14.5, result, 0.0);
    }

    @Test
    public void overdraftChargeForPremiumAccountIsHigherIfDaysOverdrawnBiggerThanOneWeek() {
        Account account = new Account(new AccountType(true), 20);

        double result = account.bankcharge();
        assertEquals(27.5, result, 0.0);
    }

}

File: csharp/07 - Demo 3 - Refactoring/Refactoring.Tests/MoveMethod/OverdraftChargeTest.cs
using Xunit;
using FluentAssertions;
using Refactoring.MoveMethod;

namespace Refactoring.Tests.MoveMethod
{
    public class FlowBetweenTest
    {

        [Fact]
        public void OverdraftCharge()
        {
            var account = new Account(new AccountType(false), 20);
            var result = account.Bankcharge();
            result.Should().Be(39.5);
        }

        [Fact]
        public void OverdraftChargeForPremiumAccountIs10()
        {
            var account = new Account(new AccountType(true), 5);
            var result = account.Bankcharge();
            result.Should().Be(14.5);
        }

        [Fact]
        public void OverdraftChargeForPremiumAccountIsHigherIfDaysOverdrawnBiggerThanOneWeek()
        {
            var account = new Account(new AccountType(true), 20);

            var result = account.Bankcharge();
            result.Should().Be(27.5);
        }
    }
}